docs/guides/directive-breakdown/readme.md

```diff
@@ -0,0 +1,151 @@
+
+* `status.storage.allocationSets.label` is an opaque string that the WLM uses when creating the spec.allocationSets entry in the DWS `Servers` resource.
+
+* `status.storage.allocationSets.minimumCapacity` is the allocation capacity in bytes. The interpretation of this field depends on the value of `status.storage.allocationSets.allocationStrategy`
+
+* `status.storage.lifetime` is used to specify how long the storage allocations will last.
+    * `job` - The allocation will last for the lifetime of the job
+    * `persistent` - The allocation will last for longer than the lifetime of the job
+
+* `status.storage.reference` is an object reference to a DWS `Servers` resource where the WLM can specify allocations
+
+### Storage Constraints
+
+Constraints on an allocation set provide additional requirements for how the storage allocations should be made on NNF nodes.
+
+* `labels` specifies a list of labels that must all be on a DWS `Storage` resource in order for an allocation to exist on that `Storage`.
+```yaml
+constraints:
+  labels:
+  - dataworkflowservices.github.io/storage=Rabbit
+  - mysite.org/pool=firmware_test
+```
+```yaml
+apiVersion: dataworkflowservices.github.io/v1alpha2
+kind: Storage
+metadata:
+  labels:
+    dataworkflowservices.github.io/storage: Rabbit
+    mysite.org/pool: firmware_test
+    mysite.org/drive-speed: fast
+  name: rabbit-node-1
+  namespace: default
+  ...
+```
+
+* `colocation` specifies how two or more allocations influence the location of each other. The colocation constraint has two fields, `type` and `key`. Currently, the only value for `type` is `exlusive`. `key` can be any value. This constraint means that the allocations from an allocation set with the colocation constraint can't be placed on an NNF node with another allocation whose allocation set has a colocation constraint with the same key. Allocations from allocation sets with colocation constraints with different keys or allocation sets without the colocation constraint are okay to put on the same NNF node.
+```yaml
+constraints:
+  colocation:
+    type: exclusive
+    key: lustre-mgt
+```
+
+* `count` this field specified the number of allocations to make when `status.storage.allocationSets.allocationStrategy` is `AllocateAcrossServers`
+```yaml
+constraints:
+  count: 5
+```
+
+* `scale` is a unitless value from 1-10 that is meant to guide the WLM on how many allocations to make when `status.storage.allocationSets.allocationStrategy` is `AllocateAcrossServers`. The actual number of allocations is not meant to correspond to the value of scale. Rather, 1 would indicate the minimum number of allocations to reach `status.storage.allocationSets.minimumCapacity`, and 10 would be the maximum number of allocations that make sense given the `status.storage.allocationSets.minimumCapacity` and the compute node count. The NNF software does not interpret this value, and it is up to the WLM to define its meaning.
```

Comment:
Matt, when does an admin create a storage profile that uses 'scale' rather than 'count'? What is the admin trying to accomplish when they do that? The webhook does not allow both to be specified together.

Reply:
`count` gives you consistent performance of the file system regardless of how big your job is. `scale` gives you file system performance that scales with the number of computes.

I think different workload types will benefit from setting the parameters differently.
 - Single shared file jobs might only need a single MDT since metadata requirements are low.
 - File per process jobs might want OSTs to scale with the number of computes.
 - Creating a persistent storage allocation would benefit from setting `count` for both OSTs and MDTs since there's no compute job, so the compute node count will be 1 (or 0 if flux adds support for jobs with no compute resources)

I think `scale` will be the more used option, but `count` is good for the situations where you want to do something specific and want full control.

Reply:
All of that sounds like useful info to pass on to the WLM developer.

Reply:
Yeah, I think you're right. I think I'll add it to the documentation for the corresponding fields in the NnfStorageProfile, though (from the year old PR). That's more of the admin/user facing documentation. That's who would be interested in knowing why you'd want to choose one value over another. The WLM documentation is more about how to take the information NNF provides in the DirectiveBreakdown and use that for scheduling computes and Rabbits.

I'm adding the contents of that old PR to the existing documentation for storage profiles. When that's done, I'll put a link to that section here.